Modal.js

```diff
@@ -0,0 +1,155 @@
+export class Modal {
+
+    #id = 'modal-version-04/13/2024'
+
+    modal = `
+        <div class="modal is-active" id="${this.#id}">
+            <div class="modal-background"></div>
+            <button class="modal-close is-large" aria-label="close"></button>
+            <div class="modal-card">
+                <header class="modal-card-head">
+                    <p class="modal-card-title"></p>
+                </header>
+                <section class="modal-card-body">
+                    <div class="field" id="modal-card-field">
+                        <label class="label"></label>
+                    </div>
+                </section>
+                <footer class="modal-card-foot is-justify-content-end">
+                    <div class="buttons">
+                        <button class="button is-danger" id="modal-primary-button"></button>
+                        <button class="button is-info" id="modal-secondary-button"></button>
+                    </div>
+                </footer>
+            </div>
+        </div> 
+    `
+
+    input = `<input id="modal-input" class="input">`
+
+    constructor(message, title, primaryBtnText, secondaryBtnText) {
+
+        this.title = title ?? 'Attention' 
+        this.message = message ?? 'This is the default alert message. Have a great day!'
+        this.requireUserInput = false
+        this.primaryBtnText = primaryBtnText ?? 'OK'
+        this.secondaryBtnText = secondaryBtnText ?? 'Cancel'
+    }
+
+    #createTemplate() {
+        
+        const existingModal = document.getElementById(`${this.#id}`)
```

Comment:
Really like how you're referencing one record of the ID!

Reply:
I liked your idea. Makes garbage collection a little more private and less prone to accidents.

Reply:
Absolutely!